ckan/lib/redis.py

```diff
@@ -0,0 +1,63 @@
+# encoding: utf-8
+
+u'''
+Redis utilities.
+
+.. versionadded:: 2.6
+'''
+
+from __future__ import absolute_import
+
+import datetime
+import logging
+
+from pylons import config
+from redis import ConnectionPool, Redis
+
+
+log = logging.getLogger(__name__)
+
+REDIS_URL_SETTING_NAME = u'ckan.redis.url'
+
+REDIS_URL_DEFAULT_VALUE = u'redis://localhost:6379/0'
```

Comment:
[We've previously settled on a warning during startup](https://github.com/ckan/ckan/pull/3165#issuecomment-232898542) in case Redis is not available, but I'm open for other solutions, including a hard fail and no default value. As far as I know, the only reason we're currently failing hard for during startup is if PostgreSQL isn't available. I've seen the Redis dependency similar to Solr, i.e. required for production but not essential. Hence a warning, but no error (just like for Solr).